src/program.ts

```diff
@@ -107,8 +109,24 @@ export function createModuleTextFromProgram(jsonObject: object): Result<string>
     function objectToString(obj: Record<string, unknown>) {
         if (obj.hasOwnProperty("@ref")) {
             const index = obj["@ref"];
-            if (typeof index === "number" && index < currentStep && Object.keys(obj).length === 1) {
-                return `step${index + 1}`;
+            const hasPath = obj.hasOwnProperty("path");
+            if (typeof index === "number" && index < currentStep && Object.keys(obj).length === (hasPath ? 2 : 1)) {
+                const path = obj["path"];
+                let str = `step${index + 1}`;
+                if (Array.isArray(path)) {
+                    for (const prop of path) {
+                        if (typeof prop === "string") {
+                            str += `.${prop}`;
+                        }
+                        else if (typeof prop === "number") {
+                            str += `[${prop}]`;
+                        }
+                        else {
+                            str += `["${prop}"]`;

```

Comment:
This part I'm not sure about. The code of the project in general is not using defensive programming, so I kept this simple. IMO, the right thing to do here is to first, have some validation to make sure that prop is safe to be used with the dot notation, otherwise fallback to the `[""]`. At the moment, this else should never be hit by code generated by LLM since it would always be either a string of prop, but it could be that the string is not a valid identifier.